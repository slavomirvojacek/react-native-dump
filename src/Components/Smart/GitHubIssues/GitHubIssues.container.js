import { Some } from '@threestup/monads'
import React from 'react'
import { compose, graphql } from 'react-apollo'

import Config from './GitHubIssues.container.config'
import GitHubIssuesComponent from './GitHubIssues.component'

const GitHubIssues = props => <GitHubIssuesComponent {...props} />

GitHubIssues.gql = Config.gql
GitHubIssues.propTypes = Config.propTypes
GitHubIssues.defaultProps = Config.defaultProps

const mapPropsToOptions = (p) => {
  // `defaultProps` are not applied at this point, therefore we have to merge explicitly
  const {repoOwner, repoName, limit} = {...Config.defaultProps, ...p}

  return {
    variables: {repoOwner, repoName, limit}
  }
}

const mapResultsToProps = ({data: {loading, error, repository, viewer}}) => ({
  issuesLoading: loading,
  error: Some(error),
  me: Some(viewer),
  issues: Some(repository)
    .map(_ => _.issues.nodes)
})

const mapAddReactionToIssueToProps = ({mutate}) => ({
  addReactionToIssue: (subjectId, content) => mutate({
    variables: {
      subjectId, content
    },
    refetchQueries: [
      'Issues'
    ]
  }).catch(console.error) // Just for demo purposes, debugging. Shouldn't be here.
})

const mapRemoveReactionFromIssueToProps = ({mutate}) => ({
  removeReactionFromIssue: (subjectId, content) => mutate({
    variables: {
      subjectId, content
    },
    refetchQueries: [
      'Issues'
    ]
  }).catch(console.error) // Just for demo purposes, debugging. Shouldn't be here.
})

export { GitHubIssues }

export default compose(
  graphql(Config.gql.Issues, {
    props: mapResultsToProps,
    options: mapPropsToOptions,
  }),
  graphql(Config.gql.AddReactionToIssue, {
    props: mapAddReactionToIssueToProps,
    options: mapPropsToOptions,
  }),
  graphql(Config.gql.RemoveReactionFromIssue, {
    props: mapRemoveReactionFromIssueToProps,
    options: mapPropsToOptions,
  })
)(GitHubIssues)
