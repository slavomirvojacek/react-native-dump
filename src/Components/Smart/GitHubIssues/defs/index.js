const userDef = (id = '') => ({id})

const reactionDef = (override = {}) => {
  const def = {
    id: '',
    content: '',
    user: userDef()
  }

  return {...def, ...override}
}

const issueDef = (override = {}) => {
  const def = {
    id: '',
    title: '',
    url: '',
    bodyText: '',
    reactions: []
  }

  return {...def, ...override}
}

export {
  userDef, reactionDef, issueDef
}
