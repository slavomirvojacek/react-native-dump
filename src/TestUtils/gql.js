const assertQueryOrMutation = (QoM) => {
  test('matches Query or Mutation', () => {
    expect(QoM).toMatchSnapshot()
  })
}

const assertGql = (Component) => {
  const { gql } = Component
  expect(gql instanceof Object).toEqual(true)
  Object
    .keys(gql)
    .forEach(k => assertQueryOrMutation(gql[k]))
}

export { assertGql }
