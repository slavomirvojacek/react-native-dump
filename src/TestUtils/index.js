export { assertGql } from './gql'
export { mockSetState } from './mocking'
export { assertSnapshots } from './snapshot'
