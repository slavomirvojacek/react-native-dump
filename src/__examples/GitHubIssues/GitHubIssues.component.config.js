import { None } from '@threestup/monads'
import PropTypes from 'prop-types'

const displayName = 'ApolloContainerView'

const propTypes = {
  issuesLoading: PropTypes.bool.isRequired,
  error: PropTypes.object.isRequired,
  issues: PropTypes.object.isRequired,
}

const defaultProps = {
  issuesLoading: false,
  error: None,
  issues: None,
}

export default {
  displayName, propTypes, defaultProps
}
