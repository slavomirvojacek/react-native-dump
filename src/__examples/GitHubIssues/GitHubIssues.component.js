import React from 'react'
import { View, Text } from 'react-native'

import Config from './GitHubIssues.component.config'
import styles from './GitHubIssues.component.styles'

import Issues from './lib/Components/Issues/Issues.component'

const IssuesLoading = () => <Text>Issues are currently loading...</Text>
const Error = ({text = 'Error'}) => <Text style={styles.error}>{text}</Text>

class GitHubIssues extends React.Component {
  someMethodWhichShouldNotBeHere = () => {
    // Whatever you want to do here can be done inside
    // the child components, e.g. <IssuesLoading />, <Error />, etc.
  }

  render () {
    const {issuesLoading, error, issues, ...rest} = this.props

    return (
      <View>
        {issuesLoading ? <IssuesLoading /> : error.match({
          some: e => <Error text={e}/>,
          none: issues.match({
            some: _ => {
              const issuesProps = {issues: _, ...rest}
              return <Issues {...issuesProps} />
            },
            none: <Text>Cannot get Issues for some reason</Text>
          })
        })}
      </View>
    )
  }
}

GitHubIssues.displayName = Config.displayName
GitHubIssues.propTypes = Config.propTypes
GitHubIssues.defaultProps = Config.defaultProps

export { GitHubIssues }
export default GitHubIssues
