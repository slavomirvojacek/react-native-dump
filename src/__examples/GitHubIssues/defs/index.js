const userDef = (id = '') => ({id})

const reactionDef = ({
                       id = '',
                       content = '',
                       user = userDef()
                     }) => ({
  id, content, user
})

const issueDef = ({
                    id = '',
                    title = '',
                    url = '',
                    bodyText = '',
                    reactions = []
                  }) => ({
  id, title, url, bodyText, reactions
})

export {
  userDef, reactionDef, issueDef
}
