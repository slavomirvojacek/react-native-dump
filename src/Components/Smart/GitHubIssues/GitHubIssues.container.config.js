import PropTypes from 'prop-types'

import gql from './GitHubIssues.graphql'

const displayName = 'GitHubIssuesContainer'

const propTypes = {
  repoOwner: PropTypes.string.isRequired,
  repoName: PropTypes.string.isRequired,
  limit: PropTypes.number.isRequired
}

const defaultProps = {
  repoOwner: 'Threestup',
  repoName: 'monads',
  limit: 20
}

export default {
  displayName, propTypes, defaultProps, gql
}
