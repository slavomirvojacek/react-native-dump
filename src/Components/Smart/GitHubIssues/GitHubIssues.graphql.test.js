import { GitHubIssues } from './GitHubIssues.container'

import { assertGql } from '../../../TestUtils/gql'

jest
  .mock('./GitHubIssues.component', () => 'GitHubIssuesComponent')

describe('GitHubIssues GraphQL Config', () => {
  assertGql(GitHubIssues)
})
