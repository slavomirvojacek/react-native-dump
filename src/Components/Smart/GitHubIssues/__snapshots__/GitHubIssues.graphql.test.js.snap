// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHubIssues GraphQL Config matches Query or Mutation 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "Issues",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "viewer",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": null,
                },
              ],
            },
          },
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "owner",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "repoOwner",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "name",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "repoName",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "repository",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "last",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "limit",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "issues",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "nodes",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "title",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "url",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "bodyText",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [
                                Object {
                                  "kind": "Argument",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "last",
                                  },
                                  "value": Object {
                                    "kind": "Variable",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "limit",
                                    },
                                  },
                                },
                              ],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "reactions",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": null,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "nodes",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "alias": null,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "content",
                                          },
                                          "selectionSet": null,
                                        },
                                        Object {
                                          "alias": null,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "user",
                                          },
                                          "selectionSet": Object {
                                            "kind": "SelectionSet",
                                            "selections": Array [
                                              Object {
                                                "alias": null,
                                                "arguments": Array [],
                                                "directives": Array [],
                                                "kind": "Field",
                                                "name": Object {
                                                  "kind": "Name",
                                                  "value": "id",
                                                },
                                                "selectionSet": null,
                                              },
                                            ],
                                          },
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "repoOwner",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "repoName",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "Int",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "limit",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 449,
    "start": 0,
  },
}
`;

exports[`GitHubIssues GraphQL Config matches Query or Mutation 2`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "AddReactionToIssue",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "input",
                },
                "value": Object {
                  "fields": Array [
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "subjectId",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "subjectId",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "content",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "content",
                        },
                      },
                    },
                  ],
                  "kind": "ObjectValue",
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "addReaction",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "reaction",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "content",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "subject",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "ID",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "subjectId",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "ReactionContent",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "content",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 237,
    "start": 0,
  },
}
`;

exports[`GitHubIssues GraphQL Config matches Query or Mutation 3`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "RemoveReactionFromIssue",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "input",
                },
                "value": Object {
                  "fields": Array [
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "subjectId",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "subjectId",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "content",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "content",
                        },
                      },
                    },
                  ],
                  "kind": "ObjectValue",
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "removeReaction",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "reaction",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "content",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "subject",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "ID",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "subjectId",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "ReactionContent",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "content",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 245,
    "start": 0,
  },
}
`;
